app/react/projects/list/index.js

```diff
@@ -9,23 +10,40 @@ import { getProjectsMerged } from '../../projects/selectors';
 import toJS from '../../utils/toJS';
 
 import * as actions from '../../projects/actions';
-import type { ProjectEntityType } from '../../types/entityTypes';
+import type { FilteredProjectEntityType } from '../../types/entityTypes';
+
+type PropTypes = {
+  fiteredProjects?: Array<FilteredProjectEntityType>,
+  filterValue?: string
+};
 
 @preload(actions.fetchProjects)
 @connect(
-  ({ projects }) => ({
-    projects: getProjectsMerged(projects)
-  }),
+  ({ projects }) => {
+    const merged = getProjectsMerged(projects);
+    return {
+      filterValue: projects.filterValue,
+      list: projects.list,
+      fiteredProjects: fuzzy.filter(projects.filterValue, merged.toJS(), { extract: el => (el.name) })
```

Comment:
Fuzzy search/filter should be inside reducer. You have declared action for projects filtering, so there are no reason why postpone processing to component.